### ISSUE TITLE
Support blog-style user guides 

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-frontend#6432]
As part of the [Homepage Hero Area project](https://docs.google.com/document/d/1PZwtYeWC2_eWBOX4a8QUBoOHeAZpEEOFRzEhpsgFrKo/edit#), we need to support static pages that have content that reads more like a blog. 

The content will be provided by @devaneymoz, it should be localized, and we don't need it to be managed via admin tools. It should just be static content that is checked into the repo. Those pages will generally link to add-on pages and collections. Design for the pages should be provided by @pwalm soon.

For URLs, I suggest something like `/guides/ad-blockers/`, `/guides/tabs/`, etc.

**Comment by eviljeff** [comment]
My vote would be for the Markdown option.  It will allow Scott to directly provide the copy including styling.  

Question: does "blog style" include images?  (3rd party images may come up against CSP or tracking blocking; AMO hosted images on the CDN is more difficult)

**Comment by jvillalobos** [comment]
There are some early drafts of the content in [this document](https://docs.google.com/document/d/1u1tbRo_ivfEAh_er2XKPciuYLARiNa2N5qXPGC0jfcY/edit?ts=5baeae48). They don't have images, but I would expect that to change in the future, so it'd be good to think of possible solutions for that.

**Comment by rebmullin** [comment]
My vote is for markdown too. I am currently looking into this and will hopefully have a POC for this soon so I can start getting feedback.

**Comment by willdurand** [comment]
For images, we could replace all image links with links pointing to a proxy like https://github.com/atmos/camo (used by GitHub). It would allow to avoid mixed content and it would be CSP friendly as we would only have one host. Processing the markdown or whatever format would be easy too.

**Comment by eviljeff** [comment]
> reference them to images in the tree in the markdown so it looks OK editing via github

Can you provide more detail on how this would work? How exactly would they be referenced in the markdown that would allow it to look good while editing?

**Comment by muffinresearch** [comment]
> Can you provide more detail on how this would work? How exactly would they be referenced in the markdown that would allow it to look good while editing?

What I'm referring to is something along the lines of this readme https://github.com/muffinresearch/reference-static-theme/blob/master/README.md

**Comment by willdurand** [comment]
Another option would by LinguiJS, which can emit PO files (using MessageFormat) and support JSX in translated content: https://lingui.js.org/tutorials/react.html#formatting.

**Comment by willdurand** [comment]
Looks like this mock-up is what we need to achieve:

![hero - article page](https://user-images.githubusercontent.com/217628/46752198-e902e880-ccbc-11e8-9810-2d6c6a49e294.png)

All pages will look the same, the number of add-ons might vary though. Those pages should not change that often (~once a month, once every 2 months) and it is mostly add-ons that would change.

**Comment by eviljeff** [comment]
> Another option would by LinguiJS, which can emit PO files (using MessageFormat) and support JSX 

JSX seems considerably more difficult to write for a non-engineer though, especially without github's built in preview that you get for Markdown.

**Comment by bobsilverberg** [comment]
We have a meeting to discuss this on Friday, but my take on it is that it looks very much like a simple templated system would work, as it's very structured and the structure is apparently unlikely to change.

As I think @willdurand mentioned in the front-end ux meeting today, it sounds like the JSX and/or markdown routes, while offering lots of flexibility, may be unnecessary to meet the requirements of this feature, and will likely be difficult both to implement, and for translators to deal with.

**Comment by pwalm** [comment]
InVision screens are here: https://mozilla.invisionapp.com/d/main/#/projects/prototypes/15640858/screens

**Comment by rebmullin** [comment]
Do you think editors would find json files easy to update? So each guide page would look something like this:

eg: privacy
```javascipt
{
  "img": "privacy.png",
  "title": "Privacy heading title",
  "sectionIntro": "Privacy ...",
  "sections": [
    {
      "title": "section 1",
      "content": "section 1 content <a href=\"/about\"> with a link</a> ...",
      "footer link": "<a href=\"/link-to-another-pg\">footer link</a>",
      "addon-guid": "1234567"
    },
    {
      "title": "section 2",
      "content": "section 2 content...",
      "footer link": "<a href=\"/link-to-another-pg\">footer link</a>",
      "addon-guid": "12345678"
    }
  ]
}
```

we can then import this into the template/component ourselves and wrap each piece with the gettext/sprintf functions. Or does this pose a problem because this is not static text?

I am still try to understand what we can /can't do with translations but would something like that work?

**Comment by willdurand** [comment]
What @rebmullin suggested above is something I wanted to build for the homepage content but I did not have the chance to work on it yet (on my free time). I wanted to build a quick&dirty create-react-app that would generate a JSON file that we could then reuse to generate code (we should not be afraid of generating code, and I love doing this), because updating the homepage content is time consuming and error-prone. I know there is a plan to get a real tool for that next year, but in the meantime, I wanted to hack onto something: quick, dirty, useful.

Anyway, I think we could also take this direction for the user guides.

I am not sure editors will contribute code (i.e. creating PRs on GitHub) anyway, so I don't know. I imagine we'll get a Google doc similar to the one for the homepage content, so everything that will help us (developers) to update the content of those pages is better.

**Comment by rebmullin** [comment]
Cool @willdurand...that’s what I was thinking … either a little create-react-app (that works as a headless cms) that can return json or just json files that live in some directory (at least in the shorter term).

Another concern was if there would be any translation issues if all the text was dynamic like this. I vaguely remember that I ran into an issue about this..

**Comment by rebmullin** [comment]
hey @jvillalobos @devaneymoz, have the guide urls been finalized? We currently have the following pages shown here:

- Stay Safe Online
- Organize Your Tabs
- Enhance Your Media Experience

REF:
https://docs.google.com/document/d/1u1tbRo_ivfEAh_er2XKPciuYLARiNa2N5qXPGC0jfcY/edit#


Can you please advise?

**Comment by devaneymoz** [comment]
@rebmullin No this content is NOT final. I'll have that content/copy finalized this week.

**Comment by rebmullin** [comment]
okay cool, thanks @devaneymoz

**Comment by rebmullin** [comment]
hey @pwalm, how should we handle these "+ Addon" buttons if the user comes across any of these pages on another browser? Should I use the (green) "Only with Firefox—Get Firefox Now" button on each card or is it better to disable these buttons and have one 1 of these green buttons somewhere?

![Alt text](https://monosnap.com/image/3R5fMJs0kPeTrT2kseL81Gc2SVzymq.png)

**Comment by pwalm** [comment]
Good question. Let me run this by the team.

**Comment by devaneymoz** [comment]
In collaboration with @MeridelW and @atsay I've [finalized content/copy](https://docs.google.com/document/d/1u1tbRo_ivfEAh_er2XKPciuYLARiNa2N5qXPGC0jfcY/edit?ts=5bbcb799) (though I'm sure once we see it in design we'll want to make slight copy teaks). 

@pwalm Could you please flow this into your latest designs when you're able?

**Comment by pwalm** [comment]
Updated:
<img width="1483" alt="screen shot 2018-11-14 at 2 51 57 pm" src="https://user-images.githubusercontent.com/6797635/48517957-dc723280-e81c-11e8-8e72-0609658ab253.png">

Button copy should read "Get Firefox" and follow the styling from the larger green button on the product page.

**Comment by MeridelW** [comment]
i agree it is confusing as-is because the CTA should be about getting the extension, not Firefox. Our longer version "Only with Firefox—Get FIrefox Now." is more helpful but I assume we don't have the room for that?  @pwalm How much room do I have? How flexible is the card in this context?

If we are this limited in space, perhaps we can do "Get the extension." The user will then discover on the journey that they need to install Firefox, which is not ideal, but it's better than "Get Firefox" alone.

**Comment by pwalm** [comment]
It's less about the card and more about the button:
<img width="870" alt="screen shot 2018-11-16 at 1 43 17 pm" src="https://user-images.githubusercontent.com/6797635/48648817-97c5d300-e9a5-11e8-8098-15bbc68462a7.png">

Doesn't look great. Any shorter copy we can use? That, or we do a message bar at the top of the page with the big green CTA and remove the buttons from the cards.

**Comment by atsay** [comment]
I like the 3rd version, with the text stacked. Any objection to that one?

**Comment by devaneymoz** [comment]
"Only with Firefox" implies exclusivity of content, doesn't it? Which wouldn't be accurate. 

What about: 

"Requires Firefox—
Get Firefox Now"

**Comment by rebmullin** [comment]
hey @pwalm @atsay @MeridelW are you okay with what @devaneymoz suggests above or did you still want to discuss this further?

**Comment by atsay** [comment]
Let's stick with the original wording and go with the stacked button (pwalm's 3rd version).

**Comment by MeridelW** [comment]
I agree with Amy about using the third, stacked option. 

Amy and I went back and forth earlier this year about the button copy and landed on the current text so I think we should stick with that for now. I agree with Scott it sort of implies exclusivity to Firefox but it's the best option for now.
